altair_pandas/_misc.py

```diff
@@ -0,0 +1,73 @@
+import altair as alt
+from typing import Union, List
+import pandas as pd
+
+toltipList = List[alt.Tooltip]
+
+
+def _preprocess_data(data):
+    for indx in ("index", "columns"):
+        if isinstance(getattr(data, indx), pd.MultiIndex):
+            setattr(
+                data,
+                indx,
+                pd.Index(
+                    [str(i) for i in getattr(data, indx)], name=getattr(data, indx).name
+                ),
+            )
+    # Column names must all be strings.
+    return data.rename(columns=str).copy()
+
+
+def scatter_matrix(
+    df,
+    color: Union[str, None] = None,
+    alpha: float = 1.0,
+    tooltip: Union[List[str], toltipList, None] = None,
+    **kwargs
+):
+    """ plots a scatter matrix
+
+    At the moment does not support neither histogram nor kde;
+    Uses f-f scatterplots instead. Interactive and with a cusotmizable
+    tooltip
+
+    Parameters
+    ----------
+    df : DataFame
+        DataFame to be used for scatterplot. Only numeric columns will be included.
+    color : string [optional]
+        Can be a column name or specific color value (hex, webcolors).
+    alpha : float
+        Opacity of the markers, within [0,1]
+    tooltip: list [optional]
+        List of specific column names or alt.Tooltip objects. If none (default),
+        will show all columns.
+    """
+    dfc = _preprocess_data(df)
+    cols = dfc._get_numeric_data().columns.tolist()
+
+    chart = (
+        alt.Chart(dfc)
+        .mark_circle()
+        .encode(
+            x=alt.X(alt.repeat("column"), type="quantitative"),
+            y=alt.X(alt.repeat("row"), type="quantitative"),
+            opacity=alt.value(alpha),
+            tooltip=tooltip or dfc.columns.tolist(),
```

Comment:
Seems like the contents of ``tooltip`` should be converted to strings in the same way the columns are within ``_preprocess_data``

Reply:
added a sanitizer for a tooltip, although there might be a better way to do that